core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/date/DateRangeAggregationBuilder.java

```diff
@@ -285,7 +281,25 @@ protected DateRangeAggregatorFactory innerBuild(SearchContext context, ValuesSou
             AggregatorFactory<?> parent, Builder subFactoriesBuilder) throws IOException {
         // We need to call processRanges here so they are parsed and we know whether `now` has been used before we make
         // the decision of whether to cache the request
-        Range[] ranges = processRanges(context, config);
+        RangeAggregator.Range[] ranges = processRanges(range -> {
+            DocValueFormat parser = config.format();
+            assert parser != null;
+            double from = range.getFrom();
+            double to = range.getTo();
+            String fromAsString = range.getFromAsString();
+            String toAsString = range.getToAsString();
+            if (fromAsString != null) {
+                from = parser.parseDouble(fromAsString, false, context.getQueryShardContext()::nowInMillis);
+            } else if (Double.isFinite(from)) {
+                from = parser.parseDouble(Long.toString((long) from), false, context.getQueryShardContext()::nowInMillis);
```

Comment:
@clintongormley I'm not sure I understand what this means for this PR. For the `date_range` aggregation the user can specify `"format" : "epoch_millis"` and then use numeric ms values in "to" and "from", regardless of whats in the mapping. The change here is that before we treated _every_ numeric value as if it was a millisecond date, no we either use the format in the mappings or in the aggregation itself, but throw an error if neither of it can parse a millisecond/second numeric date.